Python/Tests/Core.UI/IntellisenseTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestUtilities;
+using TestUtilities.UI;
+using TestUtilities.UI.Python;
+
+namespace PythonToolsUITests {
+    public class IntellisenseTests {
+        private PythonVersion Version => PythonPaths.Python37_x64 ?? PythonPaths.Python38_x64 ?? PythonPaths.Python39_x64;
+
+
+        public void AutoComplete(PythonVisualStudioApp app) {
+            EventHandler<CompletionTriggeredEventArgs> handler = null;
+            var languageName = PythonVisualStudioApp.TemplateLanguageName;
+            var slnPath = PrepareProject(app);
+
+            var project = app.OpenProject(slnPath);
+            var item = project.ProjectItems.Item("document.py");
+            var window = item.Open();
+            window.Activate();
+            var doc = app.GetDocument(item.Document.FullName);
+            try { 
+                doc.Invoke(() => {
+                    using (var e = doc.TextView.TextBuffer.CreateEdit()) {
+                        e.Insert(e.Snapshot.Length, "import sys\r\nsys.");
+                        e.Apply();
+                    }
+
+                    // Move cursor
+                    doc.TextView.Caret.MoveTo(new Microsoft.VisualStudio.Text.SnapshotPoint(doc.TextView.TextSnapshot, doc.TextView.TextSnapshot.Length));
+                });
+
+                // Wait until pylance is ready
+                Assert.IsTrue(Microsoft.PythonTools.LanguageServerClient.PythonLanguageClient.ReadyTask.Wait(60000), "Pylance did not start");
```

Comment:
in my benchmark tests I wait for a telemetry event named language_server/analysis_complete
https://dev.azure.com/mseng/DSTools/_git/pyls-benchmarks?path=%2Fdriver%2Fdrive.py&version=GBmaster&line=106&lineEnd=107&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
just a suggestion to speed things up.

Reply:
Ha!, that's exactly what the readytask waits for. I guess I picked the right event to wait for then.

Reply:
oh. nevermind. i see its using that. nice!